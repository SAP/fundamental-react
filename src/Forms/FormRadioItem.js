import 'fundamental-styles/dist/radio.css';
import FormItem from './FormItem';
import FormLabel from './FormLabel';
import PropTypes from 'prop-types';
import React from 'react';

const FormRadioItem = ({ checked, children, className, defaultChecked, disabled, id, inline, inputProps, labelProps, name, value, ...props }) => {

    return (
        <FormItem
            {...props}
            className={className}
            isInline={inline}
            key={id}>
            <FormLabel
                {...labelProps}
                disabled={disabled}>
                <input
                    {...inputProps}
                    checked={checked}
                    className='fd-radio'
                    disabled={disabled}
                    id={id}
                    name={name}
                    type='radio'
                    value={value} />
                {children}
            </FormLabel>
        </FormItem>
    );
};

FormRadioItem.displayName = 'FormRadioItem';

FormRadioItem.propTypes = {
    children: PropTypes.node.isRequired,
    checked: PropTypes.bool,
    className: PropTypes.string,
    defaultChecked: PropTypes.bool,
    disabled: PropTypes.bool,
    id: PropTypes.string,
    inline: PropTypes.bool,
    inputProps: PropTypes.object,
    labelProps: PropTypes.object,
    name: PropTypes.string,
    value: PropTypes.string
};

FormRadioItem.propDescriptions = {
    checked: 'Set to **true** when radio input is checked and a controlled component.',
    defaultChecked: 'Set to **true** when the radio input is checked and an uncontrolled component.',
    inline: 'Set to true to display a group of checkboxes in a row.',
    name: 'Sets the `name` for the radio input.',
    value: 'Sets the `value` for the radio input.'
};

export default FormRadioItem;
