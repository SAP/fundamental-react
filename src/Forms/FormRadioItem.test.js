import FormRadioItem from './FormRadioItem';
import { mount } from 'enzyme';
import React from 'react';

describe('<FormRadioItem />', () => {
    describe('Radio Item Tests', () => {
        let setup = (props) => {
            return mount(<FormRadioItem {...props}>
                Label 1
            </FormRadioItem>);
        };
        test('should add checked attribute when checked is passed to FormRadioItem', () => {
            let element = setup({
                checked: true,
                onChange: () => {}
            });

            expect(element.props().checked).toBe(true);
        });

        test('should add checked attribute when defaultChecked is passed to FormRadioItem', () => {
            let element = setup({
                defaultChecked: true,
                onChange: () => {}
            });

            expect(element.props().defaultChecked).toBe(true);
        });

        test('should set disabled to true when passed', () => {
            let element = setup({
                disabled: true
            });

            expect(element.props().disabled).toBe(true);
        });

        test('should add inline class when inline is passed', () => {
            let element = setup({
                inline: true
            });

            expect(element.find('div').hasClass('fd-form-item--inline')).toBe(true);
        });

        test('should trigger onChange FormRadioItem is clicked', () => {
            let mockCallback = jest.fn();
            let element = setup({
                onChange: mockCallback
            });

            element
                .find('input')
                .at(0)
                .simulate('change', { currentTarget: { value: 'radio-1' } });

            expect(mockCallback.mock.calls.length).toBe(1);
        });

        test('should allow props to be spread to the FormRadioItem FormItem component', () => {
            const element = mount(<FormRadioItem data-sample='Sample'>Label</FormRadioItem>);

            expect(
                element.find('div').getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
        test('should allow props to be spread to the FormRadioItem FormLabel component', () => {
            const element = mount(<FormRadioItem labelProps={{ 'data-sample': 'Sample' }}>Label</FormRadioItem>);

            expect(
                element.find('label').getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
        test('should allow props to be spread to the FormRadioItem component', () => {
            const element = mount(<FormRadioItem inputProps={{ 'data-sample': 'Sample' }}>Label</FormRadioItem>);

            expect(
                element.find('input').getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });
    test('forwards the ref', () => {
        let ref;
        class Test extends React.Component {
            constructor(props) {
                super(props);
                ref = React.createRef();
            }
            render = () => <FormRadioItem ref={ref}>foo</FormRadioItem>;
        }
        mount(<Test />);
        expect(ref.current.tagName).toEqual('INPUT');
        expect(ref.current.className).toEqual('fd-radio');
    });
});
