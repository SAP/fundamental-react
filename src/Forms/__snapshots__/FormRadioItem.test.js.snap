// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormRadioItem /> create form radio item 1`] = `
<div
  className="fd-form-item"
>
  <label
    className="fd-form-label"
  >
    <input
      className="fd-radio"
      id="radio-1"
      name="radio-group-1"
      type="radio"
      value="radio-1"
    />
    Option 1
    
  </label>
</div>
`;
